scripts/ci/build-tests.sh

```diff
@@ -0,0 +1,86 @@
+#!/usr/bin/env bash
+set -exo pipefail
+
+declare -r GOTESTSUM_VERSION=1.7.0
+
+function build_linux () {
+    local tags=osusergo,netgo,sqlite_foreign_keys,sqlite_json,static_build
+    local cc
+    case $(go env GOARCH) in
+        amd64)
+            cc=musl-gcc
+            ;;
+        arm64)
+            cc=aarch64-unknown-linux-musl-gcc
+            tags="$tags,noasm"
+            ;;
+        *)
+            >&2 echo Error: Unknown arch $(go env GOARCH)
+            exit 1
+            ;;
+    esac
+
+    local -r extld="-fno-PIC -static -Wl,-z,stack-size=8388608"
+    CGO_ENABLED=1 PKG_CONFIG=$(which pkg-config) CC=${cc} go-test-compile \
+        -tags "$tags" -o "${1}/" -ldflags "-extldflags '$extld'" ./...
+}
+
+function build_mac () {
+    CGO_ENABLED=1 PKG_CONFIG=$(which pkg-config) CC=x86_64-apple-darwin16-clang go-test-compile \
+        -tags sqlite_foreign_keys,sqlite_json -o "${1}/" ./...
+}
+
+function build_windows () {
+    CGO_ENABLED=1 PKG_CONFIG=$(which pkg-config) CC=x86_64-w64-mingw32-gcc go-test-compile \
+        -tags sqlite_foreign_keys,sqlite_json,timetzdata -o "${1}/" ./...
+}
+
+function build_test_tools () {
+    # Download gotestsum from its releases (faster than building it).
+    local -r gotestsum_url=https://github.com/gotestyourself/gotestsum/releases/download/v${GOTESTSUM_VERSION}/gotestsum_${GOTESTSUM_VERSION}_$(go env GOOS)_$(go env GOARCH).tar.gz
+    curl -L "$gotestsum_url" | tar xz -C "${1}/"
```

Comment:
we probably shouldn't use this without any verification. even though it's slower, it's probably a better idea to git clone the repo and build it, unless we can convince the gotestyourself team to add some signatures to their releases uploads.

Reply:
I see that they provide checksums. So an alternative to building it from git would be to hardcode / pin a checksum here and verify the tarball against that checksum.

Reply:
I'm going to add `gotestsum` to the cross-builder image so we don't have to grab it every time. Will do the checksum verification as part of that PR in `edge`

Reply:
Ah, I take that back; not sure it makes sense to add to the cross-builder since it's OS- and arch-specific. Will update here.